File: src/main/java/com/freya02/botcommands/api/pagination/menu/MenuBuilder.java
package com.freya02.botcommands.api.pagination.menu;

import com.freya02.botcommands.api.components.Components;
import org.jetbrains.annotations.NotNull;

import java.util.List;

/**
 * Builds a {@link Menu}
 *
 * @param <E> Type of the entries
 */
public final class MenuBuilder<E> extends BasicMenuBuilder<E, MenuBuilder<E>, Menu<E>> {
	public MenuBuilder(@NotNull Components componentsService, @NotNull List<E> entries) {
		super(componentsService, entries);
	}

	@Override
	@NotNull
	public Menu<E> build() {
		return new Menu<>(componentsService, constraints, timeout, hasDeleteButton, firstContent, previousContent, nextContent, lastContent, deleteContent, entries, maxEntriesPerPage, transformer, rowPrefixSupplier, paginatorSupplier);
	}
}

File: src/main/kotlin/io/github/freya022/botcommands/api/pagination/menu/MenuBuilder.kt
package io.github.freya022.botcommands.api.pagination.menu

import io.github.freya022.botcommands.api.core.BContext

/**
 * Builds a [Menu]
 *
 * @param E Type of the entries
 */
class MenuBuilder<E>(
    context: BContext,
    entries: List<E>
) : BasicMenuBuilder<E, MenuBuilder<E>, Menu<E>>(context, entries) {
    override fun build(): Menu<E> = Menu(context, this)
}
